src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -7423,20 +7423,20 @@ void func_8083EA94(Player* this, PlayState* play) {
     Player_AnimPlayOnce(play, this, GET_PLAYER_ANIM(PLAYER_ANIMGROUP_throw, this->modelAnimType));
 }
 
-s32 func_8083EAF0(Player* this, Actor* actor) {
-    if ((actor != NULL) && !(actor->flags & ACTOR_FLAG_23) &&
+s32 Player_CanThrowCarriedActor(Player* this, Actor* actor) {
+    if ((actor != NULL) && !(actor->flags & ACTOR_FLAG_THROW_ONLY) &&
```

Comment:
I guess it kinda is, though its pretty obviously just a safety null check before dereferencing it to check the flag. And in both cases its used they are blocked by safety null checks anyway

Reply:
Yea it's a weird situation. If the intent is that the function should never be passed in a NULL actor, then the `actor !=NULL` check isn't needed. If the test is as you say it is, it would seem like allowing a NULL pointer to be tossed would be a bug.

I feel it needs a comment, and would get behind the idea that `actor` being passed in should just not be NULL in spite of the NULL check.

Reply:
I agree a short comment about this seems good. Could also use the opportunity to comment that returning false means "can drop instead" (rather than the object being glued to link's hands or something)

Reply:
I wrote something but it isnt very short. Not sure how to make it shorter:
```c
    // There is a safety check on the `actor` argument to prevent a null dereference when checking the flag.
    // However if a NULL actor pointer is passed, it will result in "true" being returned.
    // It doesn't make sense for a non-existent actor to be thrown or dropped, so the return wouldn't
    // make sense either way.
    // Instead, it is safer to prevent a NULL actor pointer from being passed to this function in the first place
    if ((actor != NULL) && !(actor->flags & ACTOR_FLAG_THROW_ONLY) &&
        ((this->speedXZ < 1.1f) || (actor->id == ACTOR_EN_BOM_CHU))) {
        return false;
    }
    ```

Reply:
I'd put it on Player_CanThrowCarriedActor instead, maybe something like:
```
// Checks if Link should throw or drop the carried actor if he stops carrying it.
// Returns true if Link should throw the carried actor (or if the carried actor is NULL),
// and returns false if he should drop it instead.
```
I'm not sure it's worth elaborating more

Reply:
That phrasing makes it sound intentional when I dont think it really is

Reply:
works for me

Reply:
Pushed. I had to change the function comment a bit, because the function itself is not aware that its actually the actor being carried. Even though it is the assumption.

Reply:
@mzxrules please verify this works for you so I can merge